src/nvim/lua/vim.lua

```diff
@@ -1,3 +1,46 @@
+local function system(cmd)
+  local out = vim.api.nvim_call_function('system', { cmd })
+  local err = vim.api.nvim_get_vvar('shell_error')
+  local args_str = vim.api.nvim_call_function('string', { cmd })
+  if 0 ~= err then
+    error('command failed: '..args_str)
+  end
+  return out
+end
+
+local function systemlist(...)
```

Comment:
@neovim/core I'm considering documenting these in `:help lua` as part of the `vim` module. The normal lua idiom would be to return `status, rv` pair, where `status=false` on failure. But I think it's really useful to show the failed command with `error()`. Thoughts?